doc/services/llext/index.rst

```diff
@@ -22,3 +22,5 @@ and introspected to some degree, as well as unloaded when no longer needed.
 
    The LLEXT subsystem requires architecture-specific support. It is currently
    available only on RISC-V, ARM, ARM64, ARC (experimental) and Xtensa cores.
+   Harvard architecture cores that separate code and data paths and have no
+   common memory are not supported.
```

Comment:
Just a comment: for ARC with HARVARD (which in most of the cases means ICCM + DCCM usage)  it's really easy to overcome this by loading to ICCM directly. moreover CCMs are uncached (cause they are already fast enough) - so we won't even need to have caches sync.

Reply:
Sure, but neither LLEXT nor our linker scripts are not exactly ready for that at the moment. Besides, it could be quite wasteful to load the whole ELF into ICCM (LLEXT_STORAGE_WRITABLE=y case) or to place the whole LLEXT heap into ICCM (LLEXT_STORAGE_WRITABLE=n case). Do our real-world Harvard configurations have enough ICCM to make it practical? If they do, we may prioritize such work to make it happen.

Reply:
Right, I guess ideally we would need split heaps for this sort of architecture am I understanding this right? There's other reasons to consider this as well, like avoiding needing all of user mode for memory permission bits.

Reply:
> Right, I guess ideally we would need split heaps for this sort of architecture am I understanding this right? There's other reasons to consider this as well, like avoiding needing all of user mode for memory permission bits.

Yes, I think we should at least consider splitting the heap into executable and non-executable parts.